compiler/modules/types.emit/emit.bal

```diff
@@ -0,0 +1,186 @@
+import wso2/nballerina.err;
+import wso2/nballerina.types as t;
+import nballerina.types.bdd;
+
+final readonly & string[] RO_ATOMS = [
+    "()", "boolean", "int", "float", "decimal", "string", 
+    "error", "function", "typedesc", "handle"];
+
+final readonly & string[] RW_ATOMS = [
+    "xml", "list", "map", "table", "object", "stream", "future"
+];
+
+# Serialize SemType into pseudo-Ballerina type syntax
+#
+# + env - Type environment  
+# + ty - Type to be serialized
+# + return - String representation  
+public function semTypeToString(t:Env env, t:SemType ty) returns string {
+    if ty is t:UniformTypeBitSet {
+        return nonComplexSemTypeToString(ty);
+    } 
+    else {
+        return complexSemTypeToString(ty);
+    }
+}
+
+function nonComplexSemTypeToString(t:UniformTypeBitSet ty) returns string {
+    // JBUG should work without ether casts 
+    if <int>ty == <int>t:NEVER {
```

Comment:
The casts are ugly here.

I changed types to declare t:NEVER as a UniformTypeBitSet rather than a SemType, so you shouldn't need casts (although jBallerina has some bugs here).